core/app/models/spree/order.rb

```diff
@@ -686,9 +687,9 @@ def order_total_after_store_credit
 
     def total_applicable_store_credit
       if can_complete? || complete?
-        payments.store_credits.valid.sum(:amount)
+        valid_store_credit_payments.to_a.sum(&:amount)
```

Comment:
Like above, I think the previous version was more performant, as it did all calculations in the DB without instantiating and cycling ruby objects

Reply:
It is not more performant this way. Quite contrary this only does the math in the database if the payments have not been loaded yet. If they already were loaded the sum in database approach is actually slower, as it unnecessarily loads the payments again.